tests/unit/base_tests_zarrio.py

```diff
@@ -502,60 +567,79 @@ def test_write_attributes_write_3Darray_of_floats(self):
     def test_write_attributes_write_reference_to_datasetbuilder(self):
         data_1 = np.arange(100, 200, 10).reshape(2, 5)
         dataset_1 = DatasetBuilder('dataset_1', data_1)
-        testgroup = self.io._ZarrIO__file  # For testing we just use our file and create some attributes
+        tempIO = ZarrIO(self.store, mode='w')
+        tempIO.open()
+        testgroup = tempIO.file  # For testing we just use our file and create some attributes
         attr = {'attr1': dataset_1}
-        self.io.write_attributes(testgroup, attr)
+        tempIO = ZarrIO(self.store, mode='w')
```

Comment:
This confuses me. First, open the zarr store and get the file object. Then open it again and write some attributes. Why open it twice?

Reply:
I don't think it is necessary. I updated the test case to avoid opening the file twice.